mintlify/snippets/kyc-onboarding.mdx

```diff
@@ -0,0 +1,632 @@
+</Note>
+
+### Generate KYC Link
+
+<RequestExample>
+```http GET /customers/kyc-link
+GET /customers/kyc-link?redirect_uri=https://yourapp.com/onboarding-complete
+Authorization: Basic <base64-encoded-credentials>
+```
+</RequestExample>
+
+<ResponseExample>
+```json Success
+{
+  "kycUrl": "https://kyc.lightspark.com/onboard/abc123def456"
+}
+```
+</ResponseExample>
+
+<ParamField query="redirect_uri" type="string">
+Optional URI parameter that specifies where the customer will be automatically redirected after successfully completing the hosted KYC verification process. If not provided, customers will be redirected to a default completion page.
+</ParamField>
+
+<ResponseField name="kycUrl" type="string" required>
+A hosted KYC link for your customer to complete identity verification and onboarding. This URL contains the KYC session and any redirect configuration.
+</ResponseField>
+
+<Note>
+The `redirect_uri` parameter is embedded in the KYC URL and will be used to redirect the customer back to your application after they complete the verification process. This provides a seamless user experience by returning them to your platform.
+</Note>
+
+### Complete KYC Process
+
+<Steps>
+<Step title="Generate KYC Link">
+  Call the `/customers/kyc-link` endpoint with your `redirect_uri` parameter to generate a hosted KYC URL for your customer.
+  
+  <Note>
+  The `redirect_uri` parameter is embedded in the generated KYC URL and will be used to automatically redirect the customer back to your application after they complete verification.
+  </Note>
+</Step>
+
+<Step title="Redirect Customer">
+  Redirect your customer to the returned `kycUrl` where they can complete their identity verification in the hosted interface.
+  
+  <Warning>
+  The KYC link is single-use and expires after a limited time period for security.
+  </Warning>
+</Step>
+
+<Step title="Customer Completes Verification">
+  The customer completes the identity verification process in the hosted KYC interface, providing required documents and information.
+  
+  <Info>
+  The hosted interface handles document collection, verification checks, and compliance requirements automatically.
+  </Info>
+  
+  <Note>
+  After verification processing, you'll receive a KYC status webhook notification indicating the final verification result.
+  </Note>
+</Step>
+
+<Step title="Automatic Redirect">
+  Upon successful KYC completion, the customer is automatically redirected to your specified `redirect_uri` URL.
+  
+  <Check>
+  The customer account will be automatically created by the system upon successful KYC completion. You can identify the new customer using your `platformCustomerId` or other identifiers.
```

Comment:
The customer object is actually created immediately, right? Not just after successful kyc. Maybe we should return the new customer ID somewhere as a param to the redirect URI or something so that it's queryable? I don't think they can use the platformCustomerId unless they give us one when creating the KYC link, right?